h2/src/main/org/h2/bnf/context/DbSchema.java

```diff
@@ -71,11 +71,8 @@
         } else if (contents.isPostgreSQL() &&
                 StringUtils.toUpperEnglish(name).startsWith("PG_")) {
             isSystem = true;
-        } else if (contents.isDerby() && name.startsWith("SYS")) {
-            isSystem = true;
-        } else {
-            isSystem = false;
-        }
+        } else
+            isSystem = contents.isDerby() && name.startsWith("SYS");
```

Comment:
I also don't think that this change and similar to it are good. Because they introduce mixed logic that complicates reading of sources. If you want change this method to save a few bytes in compiled bytecode, it's better to reorganize all these if-else branches into something like
```Java
isSystem =
    // comment1
    (condition1)
    // comment2
    || (condition2 && condition3)
    // comment3
    || (condition4 && (condition5)
..... 
```
If this is not possible in some case due to complex conditions, it's better to leave this code as is.

Reply:
is the current version alright? I can't see the mixed logic here - it's just avoiding using additional if.

Reply:
The problem here is that with this change case with Derby looks like “default” / “other” case. But this condition is not special and should be treated like conditions above it. But this is not clear for casual reader.

This only reduces bytecode size by two instructions and probably saves nothing after JIT compilation with C2. Readability is more important than this.

But the whole if-else if-else operator may be reorganized to avoid all these `= true` and `= false`, but only with proper line breaks and indentation to preserve readability. This reduces size of bytecode significantly. But this also is not necessary because this constructor is not performance-critical.

Reply:
I applied your suggestion and removed inlined all ifs.